js/start.js

```diff
@@ -253,27 +256,167 @@ function onboardIfNeeded() {
   return onboardIfNeededDeferred.promise();
 }
 
-// let's start our flow
-fetchConfig().done((data) => {
-  app.profile = new Profile({ id: data.guid });
-
-  app.settings = new Settings();
-  // We'll default our server language to whatever is stored locally.
-  app.settings.set('language', app.localSettings.get('language'));
-
-  // Beyond the start-up flow in this file, any language changes should ideally
-  // be done via a save on a clone of the app.settings model. When the save succeeds,
-  // update the app.settings model which will in turn update our local
-  // settings model. You shouldn't be directly updating the language in our local
-  // settings model.
-  app.settings.on('change:language', (settingsMd, lang) => {
-    app.localSettings.save('language', getValidLanguage(lang));
-  });
+ // let's start our flow - do we need onboarding?,
+ // fetching app wide models...
+function start() {
+  fetchConfig().done((data) => {
+    app.profile = new Profile({ id: data.guid });
+
+    app.settings = new Settings();
+    // We'll default our server language to whatever is stored locally.
+    app.settings.set('language', app.localSettings.get('language'));
+
+    // Beyond the start-up flow in this file, any language changes should ideally
+    // be done via a save on a clone of the app.settings model. When the save succeeds,
+    // update the app.settings model which will in turn update our local
+    // settings model. You shouldn't be directly updating the language in our local
+    // settings model.
+    app.settings.on('change:language', (settingsMd, lang) => {
+      app.localSettings.save('language', getValidLanguage(lang));
+    });
 
-  onboardIfNeeded().done(() => {
-    app.pageNav.navigable = true;
-    app.loadingModal.close();
-    location.hash = location.hash || app.profile.id;
-    Backbone.history.start();
+    onboardIfNeeded().done(() => {
+      app.pageNav.navigable = true;
+      app.loadingModal.close();
+      location.hash = location.hash || app.profile.id;
+      Backbone.history.start();
+    });
   });
+}
+
+// connect to the API websocket
+// todo: this will be incorporated in the server
+// connection flow
+let socketOpened = false;
+let lostSocketConnectionDialog;
+
+app.apiSocket = new Socket(app.getSocketUrl());
+app.apiSocket.on('open', () => {
+  if (!socketOpened) {
+    socketOpened = true;
+    if (lostSocketConnectionDialog) lostSocketConnectionDialog.remove();
+    lostSocketConnectionDialog = null;
+    start();
+  } else {
+    location.reload();
+  }
+});
+
+app.apiSocket.on('close', () => {
+  if (lostSocketConnectionDialog) return;
+
+  lostSocketConnectionDialog = new Dialog({
+    title: 'Socket connection failure',
```

Comment:
Didn't translate because this dialog is very temporary. Once we have the connection modal in place, I think we would just want to show that upon socket failures (at least that's what we do in v1).